### ISSUE TITLE
Support autocast calls in dynamo

### ISSUE THREAD

**Issue by voznesenskym** [pytorch/torchdynamo#1263]
ATM, we get:

```
AssertionError: torch.* op returned non-Tensor [autocast] call_function <class 'torch.amp.[autocast]'>
```

**Comment by jansel** [comment]
Should be similar to no_grad()

**Comment by voznesenskym** [comment]
> Should be similar to no_grad()

Yeah I have it in a local branch I need to clean up and push. There's a little distraction I feel into refactoring how we do contexts, grad was our first one and it's a little too woven into some of our existing context classes.

**Comment by voznesenskym** [comment]
https://github.com/pytorch/torchdynamo/pull/1289
